__tests__/changelog-updater.test.ts

```diff
@@ -364,19 +364,80 @@ test('Updates existing section when between other sections', async () => {
 - Added a new feature
 
 ### Dependencies
-- Bumps \`other-package\` from v2 to v3
-- Bumps \`package\` from v1 to v2
+- Bump \`other-package\` from v2 to v3
+- Bump \`package\` from v1 to v2
 
 ### Removed
 - Removed a feature
 
 ## [v0.9.0]
 
 ### Dependencies
-- Bumps \`package\` from alpha to v1`
+- Bump \`package\` from alpha to v1`
   )
 })
 
+const CHANGELOG_WITH_PROPER_SECTIONS_AND_ENTRIES_DIFFERENT_PREFIX = `# Changelog
+
+## [v1.0.0]
+### Dependencies
+- Bump \`different-package\` from v1 to v2`
+
+test('adds an entry with a different prefix to the changelog when section already exists with entry', async () => {
+  mockReadStream(CHANGELOG_WITH_PROPER_SECTIONS_AND_ENTRIES)
+
+  await updateChangelog(PACKAGE_ENTRY, 'v1.0.0', './CHANGELOG.md', 'Update')
+
+  const params = fs.writeFileSync.mock.calls[0]
+  expect(params[0]).toStrictEqual('./CHANGELOG.md')
+  expect(params[1]).toStrictEqual(`# Changelog
+
+## [v1.0.0]
+### Dependencies
+- Bump \`different-package\` from v1 to v2
+- Update \`package\` from v1 to v2`)
+})
+
+const CHANGELOG_WITH_DUPLICATE_ENTRY_DIFFERENT_PREFIX = `# Changelog
+
+## [v1.0.0]
+### Dependencies
+- Bump \`package\` from v1 to v2`
+
+test('keeps prefix on entry with a different prefix but is otherwise a duplicate', async () => {
+  mockReadStream(CHANGELOG_WITH_DUPLICATE_ENTRY_DIFFERENT_PREFIX)
+
+  await updateChangelog(PACKAGE_ENTRY, 'v1.0.0', './CHANGELOG.md', 'Update')
+
+  const params = fs.writeFileSync.mock.calls[0]
+  expect(params[0]).toStrictEqual('./CHANGELOG.md')
+  expect(params[1]).toStrictEqual(`# Changelog
+
+## [v1.0.0]
+### Dependencies
+- Bump \`package\` from v1 to v2`)
```

Comment:
Note there is some subtlety here regarding finding existing entries/duplicates. I've chosen _not_ to change the prefix word of an existing entry if a different prefix word is provided.